### PR TITLE
Add JS for chapter of computational complexity

### PR DESCRIPTION
**Pull request by gyt95**

### CODE REVIEW

**Review by justin-tse — COMMENTED** [review]
Hi! @gyt95 Thanks for the contribution! This PR will keep open waiting for the space_time_tradeoff.md and leetcode_two_sum.js update. There are some things that need you to update this PR:
1. You didn't use ';' at the end of every line in this PR. Could you update them to match other JS codes, at the end of every line use ';'.
2. Our repositories' indentation style is 4 spaces, please update them to match other JS codes.

**Comment by gyt95** [comment]
> Hi! @gyt95 Thanks for the contribution! This PR will keep open waiting for the space_time_tradeoff.md and leetcode_two_sum.js update. There are some things that need you to update this PR:
> 
> 1. You didn't use ';' at the end of every line in this PR. Could you update them to match other JS codes, at the end of every line use ';'.
> 2. Our repositories' indentation style is 4 spaces, please update them to match other JS codes.

Ok, I made a new commit, you can take a look.

**Comment by gyt95** [comment]
> Hi! @gyt95 Thanks for the contribution! This PR will keep open waiting for the space_time_tradeoff.md and leetcode_two_sum.js update. There are some things that need you to update this PR:
> 
> 1. You didn't use ';' at the end of every line in this PR. Could you update them to match other JS codes, at the end of every line use ';'.
> 2. Our repositories' indentation style is 4 spaces, please update them to match other JS codes.

Ok, I made two new commits, you can take a look.

**Review by justin-tse — COMMENTED** [review]
Hi!@gyt95  The code should have a test and pass the test, please follow the Java code to make the JS test.
You could see the Contribution Guideline: https://github.com/krahets/hello-algo/issues/15#issue-1464069591

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * @Author: gyt95 (gytkwan@gmail.com)
  * @Date: 2022-12-15 10:51:54 
  * @Last Modified by: gyt95 (gytkwan@gmail.com)
```
Comment (krahets) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:4]: Please set the file header following the format of other JS files.

Comment (gyt95) [reply codes/javascript/chapter_computational_complexity/leetcode_two_sum.js]: I think it should work this time.

Comment (gyt95) [reply codes/javascript/chapter_computational_complexity/leetcode_two_sum.js]: I forgot something in test, wait a minute.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/**
+ * File: leetcode_two_sum.js
+ * Created Time: 2022-12-15
+ * Author: gyt95 (gytkwan@gmail.com)
+ */
+
+/**
+ * @param {number[]} nums
```
Comment (krahets) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:8]: We don't need this function comment, please remove it

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * @param {number} target
+ * @return {number[]}
+ */
+function twoSumBruteForce(nums, target) {
+    let n = nums.length;
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:8]: Here maybe you could consider using const and not let. Feel free to change it.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    // 两层循环，时间复杂度 O(n^2)
+    for (let i = 0; i < n; i++) {
+        for (let j = i + 1; j < n; j++) {
+            if (nums[i] + nums[j] === target) {
+                return [i, j];
+            }
+        }
+    }
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:16]:
```suggestion
    }
    return [];
```
You need to consider when we can not find the result.
We should follow the Java code to translate.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+function twoSumHashTable(nums, target) {
+    // 辅助哈希表，空间复杂度 O(n)
+    let m = {};
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:21]: Here maybe you could consider using const and not let. Feel free to change it.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    // 单层循环，时间复杂度 O(n)
+    for (let i = 0; i < nums.length; i++) {
+        if (m[nums[i]] !== undefined) {
+            return [m[nums[i]], i];
+        } else {
+            m[target - nums[i]] = i;
+        }
+    }
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:29]:
```suggestion
    }
    return [];
```
You need to consider when we can not find the result.
We should follow the Java code to translate.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+/* Driver Code */
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:32]:
```suggestion
/* Driver Code */
// 方法一
```

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+let nums = [2, 7, 11, 15], target = 9;
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:33]: Here maybe you could consider using const and not let. Feel free to change it.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js (hunk 2 above)
Comment (gyt95) [reply codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:8]: ok

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+twoSumBruteForce(nums, target)
+console.log("使用暴力枚举后得到结果：", nums)
+
+let nums1 = [2, 7, 11, 15], target1 = 9;
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:37]: Do not need to define the test case again.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        } else {
+            m[target - nums[i]] = i;
+        }
+    }
+}
+
+/* Driver Code */
+let nums = [2, 7, 11, 15], target = 9;
+twoSumBruteForce(nums, target)
+console.log("使用暴力枚举后得到结果：", nums)
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:35]:
```suggestion
let res = twoSumBruteForce(nums, target)
console.log("方法一 res = ", res)
```

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+let nums1 = [2, 7, 11, 15], target1 = 9;
+twoSumHashTable(nums1, target1)
+console.log("使用辅助哈希表后得到结果", nums1)
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:39]: Please follow the Java code to test the result and not the nums.

File: codes/javascript/chapter_computational_complexity/leetcode_two_sum.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    }
+}
+
+/* Driver Code */
+let nums = [2, 7, 11, 15], target = 9;
+twoSumBruteForce(nums, target)
+console.log("使用暴力枚举后得到结果：", nums)
+
+let nums1 = [2, 7, 11, 15], target1 = 9;
+twoSumHashTable(nums1, target1)
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/leetcode_two_sum.js:38]:
```suggestion
// 方法二
twoSumHashTable(nums1, target1)
```

**Review by justin-tse — COMMENTED** [review]
Hi! @gyt95, please update the code and the docs. Thank you for your contribution.

**Review by justin-tse — APPROVED** [review]
Good job! Thank you.

**Review by krahets — APPROVED** [review]

**Pull request merged**